pkg/cloud/services/eks/tags.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eks
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/converters"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/tags"
+)
+
+func (s *Service) reconcileTags(cluster *eks.Cluster) error {
+	clusterTags := converters.MapPtrToMap(cluster.Tags)
+	buildParams := s.getEKSTagParams(*cluster.Arn)
+	tagsBuilder := tags.New(buildParams, tags.WithEKS(s.EKSClient))
+	if err := tagsBuilder.Ensure(clusterTags); err != nil {
+		return fmt.Errorf("failed ensuring tags on cluster: %w", err)
+	}
+
+	return nil
+}
+
+func (s *Service) getEKSTagParams(id string) *infrav1.BuildParams {
+	name := fmt.Sprintf("%s-eks-cp", s.scope.Name())
+
+	return &infrav1.BuildParams{
+		ClusterName: s.scope.Name(),
```

Comment:
Same comment as https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1815#discussion_r469903727 wrt to converting kubernetes resource name to the eks resource name

Reply:
If the object name isn't going to equal the EKS cluster name, wdyt about also adding a status field with the resolved name in it? That way I can consume that from the EKS bootstrap provider.

Reply:
Ha, we where thinking along exactly the same lines. I have added this to the status:
https://github.com/richardcase/cluster-api-provider-aws/blob/ekscontrolplane/exp/api/v1alpha3/awsmanagedcontrolplane_types.go#L124

Reply:
I have changed the code to adopt the naming convention